File: core/runtime/semantic_function_registry.py
from __future__ import annotations

from typing import Any, Callable, Dict


class SemanticFunctionRegistry:
    def __init__(self) -> None:
        self._functions: Dict[str, Callable[..., Any]] = {}

    def register(
        self,
        name: str,
        fn: Callable[..., Any],
    ) -> None:

        self._functions[name] = fn

    def call(
        self,
        name: str,
        *args: Any,
        **kwargs: Any,
    ) -> Any:

        return self._functions[name](
            *args,
            **kwargs,
        )

File: src/runtime/semanticFunctionRegistry.ts
/**
 * Converted from Python: core/runtime/semantic_function_registry.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "./pyCompat.js";

export class SemanticFunctionRegistry {
  declare _functions: any;
  constructor() {
    this._functions = {};
  }
  register(name: any, fn: any): any {
    py.setItem(this._functions, name, fn);
  }
  call(name: any, ...args: any[]): any {
    var kwargs: Record<string, any> = {};
    return py.at(this._functions, name)(...py.iter(args), kwargs);
  }
}
